File: src/Server/WebApi/Dtos/BookDto.cs
namespace Cbc.WebApi.Dtos;

using Cbc.WebApi.Interfaces;
using Cbc.WebApi.Models.Entities;

public class BookDto : IMapFrom<Book>
{
    public Guid Id { get; set; }

    public string Title { get; set; } = null!;

    public string Author { get; set; } = null!;

    public string? Description { get; set; }

    public string? UserEmail { get; set; } = null!;

    public string? ThumbnailLink { get; set; }

    public string? Isbn { get; set; }

    public int? PageCount { get; set; }

    public UserSimpleDto? User { get; set; }

    public List<BookRatingDto>? Ratings { get; set; }
}

File: src/Client/src/app/api/models/book-dto.ts
/* tslint:disable */
/* eslint-disable */
import { BookRatingDto } from '../models/book-rating-dto';
import { UserSimpleDto } from '../models/user-simple-dto';
export interface BookDto {
  author: string;
  description?: string | null;
  id: string;
  isbn?: string | null;
  pageCount?: number | null;
  ratings?: Array<BookRatingDto> | null;
  thumbnailLink?: string | null;
  title: string;
  user?: UserSimpleDto | null;
  userEmail?: string | null;
}
